test/compile_test.sh

```diff
@@ -2,92 +2,93 @@
-  assertCaptured "Decrypting foo3.ejson --> foo3.json"
-  assertCaptured "Decrypting foo4.ejson --> foo4.json"
-  assertCaptured "Done. Decrypted 5 different ejson file(s)"
+    capture_profile_d_export_script
+    assertCapturedSuccess
+    assertCaptured "Decrypting EJSON file at \`config.ejson\` and exporting as environment variables"
+    assertEquals "bar" "$foo"
+    assertEquals "" "$_public_key"
+  )
 }
 
-test_environment_set() {
-  compile_with_fixture environment_set
-  assertCapturedSuccess
-  assertCaptured 'EJSON_ENVIRONMENT is `production`; enumerating and decrypting EJSON files of the form `*.production.ejson`'
-  assertCaptured 'Decrypting config.production.ejson --> config.json'
-  assertCaptured 'Done. Decrypted 1 different ejson file(s)'
+test_missing_private_key() {
+  (
+    compile_with_fixture missing_private_key
+    assertCapturedError
+    assertCaptured "Loading keypair from environment variables"
+    assertCaptured 'EJSON_PRIVATE_KEY is undefined; make sure EJSON_PRIVATE_KEY and EJSON_FILE are set'
+  )
 }
 
-test_environment_unset() {
-  compile_with_fixture environment_unset
-  assertCapturedSuccess
-  assertCaptured 'EJSON_ENVIRONMENT is undefined or empty; enumerating and decrypting EJSON files of the form `*.ejson`'
-  assertCaptured "Decrypting config.ejson --> config.json"
-  assertCaptured "Done. Decrypted 1 different ejson file(s)"
+test_missing_ejson_file() {
+  (
+    compile_with_fixture missing_ejson_file
+    assertCapturedError
+    assertCaptured "Loading keypair from environment variables"
+    assertCaptured 'EJSON_FILE is undefined; make sure EJSON_PRIVATE_KEY and EJSON_FILE are set'
+  )
 }
 
-test_overwrite() {
-  compile_with_fixture overwrite
-  assertCapturedSuccess
-  assertCaptured 'EJSON_ENVIRONMENT is `production`; enumerating and decrypting EJSON files of the form `*.production.ejson`'
-  assertCaptured 'Decrypting config.production.ejson --> config.json [overwriting existing file]'
+test_ejson_file_not_found_in_build_dir() {
+  (
+    compile_with_fixture missing_ejson_file_in_build
+    assertCaptured "Loading keypair from environment variables"
+    capture_profile_d_export_script
+    assertCapturedError
+    assertCaptured "EJSON_FILE could not be found at config.ejson"
+  )
 }
 
-test_overwrite_directory() {
-  compile_with_fixture overwrite_directory
-  assertCapturedError
-  assertCaptured 'EJSON_ENVIRONMENT is `production`; enumerating and decrypting EJSON files of the form `*.production.ejson`'
-  assertCaptured 'config.json: already exists but is not a regular file'
+test_bad_keypair() {
+  (
+    compile_with_fixture bad_keypair
+    assertCaptured "Loading keypair from environment variables"
+    capture_profile_d_export_script
+    assertCaptured "Decrypting EJSON file at \`config.ejson\` and exporting as environment variables"
+    assertCapturedError
+    assertCaptured "Decryption failed: couldn't decrypt message"
+  )
```

Comment:
Should we add a test for invalid JSON? I know from using `jq` on the CLI it doesn't output the full file to output to stdout but it'd be nice to capture that in a test.

Reply:
How could we test that without stubbing the `ejson` call? We could test an ejson file with invalid json

Reply:
yeah just have a test fixture that decrypts but is invalid JSON

Reply:
I don't think it's worth it. That kind of test belongs in EJSON itself. That scenario results in `Decryption failed: unexpected end of JSON input` and a non-zero exit code, which will work fine with this buildpack (we wouldn't need to do anything different it's just another error from ejson).